python/README.rst

```diff
@@ -1,26 +1,18 @@
 DOLFINx Python interface
```

Comment:
The content of this file will be replaced with the one coming from #2707 anyways.